src/btree/bt_debug.c

```diff
@@ -1185,10 +1212,21 @@ __debug_page_col_fix(WT_DBG *ds, WT_REF *ref)
 
     if (dsk != NULL) {
         ins = WT_SKIP_FIRST(WT_COL_UPDATE_SINGLE(page));
-        WT_FIX_FOREACH (btree, dsk, v, i) {
+        curtw = 0;
+        numtws = WT_COL_FIX_TWS_SET(page) ? page->pg_fix_numtws : 0;
+
+        WT_COL_FIX_FOREACH_BITS (btree, dsk, v, i) {
             WT_RET(ds->f(ds, "\t%" PRIu64 "\t{", recno));
             WT_RET(__debug_hex_byte(ds, v));
-            WT_RET(ds->f(ds, "}\n"));
+            WT_RET(ds->f(ds, "}"));
+            if (curtw < numtws && recno - ref->ref_recno == page->pg_fix_tws[curtw].recno_offset) {
```

Comment:
I don't follow -- it is iterating through the bits, but also going through the time windows (and the update list), so if the current time window matches the row we're looking at, it prints it after the bit value and advances to the next one. Otherwise, the time window belongs to some later row so it doesn't touch it.
(I'm pretty sure this code works as I've been using it for some time now)

Reply:
Yes, you are correct. I didn't clearly observe that it increments only when the recno matches.